### PR TITLE
feat(cache): add warning and confirmation to cache:clear:all command

### PR DESCRIPTION
**Pull request by SpiGAndromeda**
### 1. Why is this change necessary?

The `cache:clear:all` command deletes all kernel cache directories except the currently active one. When CLI and web server use different plugin configurations (producing different cache hashes), running this command can cause 500 errors on running web instances. Users should be aware of this risk before proceeding.

### 2. What does this change do, exactly?

- Display a warning explaining the potential impact before execution
- Prompt for confirmation in interactive terminals (TTY)
- Add `--force` flag to skip confirmation
- Skip confirmation automatically in non-interactive environments (cron, CI, scripts) for backward compatibility
- Add `TtyDetector` service for testable TTY detection

### 3. Describe each step to reproduce the issue or behaviour.

1. Run `bin/console cache:clear:all` in a terminal
2. See warning about deleting other kernel cache directories
3. Confirm with "yes" or abort with "no"
4. Or use `--force` to skip: `bin/console cache:clear:all --force`

### 4. Please link to the relevant issues (if any).

Documents risk of cache:clear:all for mixed environments (#4545)

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

### CODE REVIEW

**Comment by SpiGAndromeda** [comment]
> Warnings
> ⚠️ 	Please be kind and add unit tests for your new code in these files:
> 
> src/Core/Framework/Adapter/Cache/ContainerCacheDirectoryClearer.php
> 
> If you are sure everything is fine with your changes, you can resolve this warning.
> You can run `composer make:coverage` to generate dummy unit tests for files that are not covered

The changes are covered by integration tests. The added source code is "only" one class with one line of delegation, so a unit test is unnecessary here.

**Comment by SpiGAndromeda** [comment]
I decided to go another way.

The `cache:clear` command keeps its current behvior. The differences between the `cache:clear` and `cache:clear:all` command are documented (see: https://github.com/shopware/docs/pull/2088).

A confirmation is added to the `cache:clear:all` command with a warning that this might impact running requests and lead to 500 errors. The confirmation will only be prompted in interactive environments. So the behavior in envirements like crontab doesnt change.

File: src/Core/Framework/Adapter/Command/CacheClearAllCommand.php [hunk 1]
```diff
@@ -42,6 +55,20 @@ protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $io = new ShopwareStyle($input, $output);
 
+        $io->warning([
+            'This command will delete all kernel cache directories except the currently active one.',
+            'If a web server is running with a different plugin configuration, this may cause 500 errors.',
+            'In clustered environments or when web and CLI use different plugins, consider using "cache:clear" instead.',
+        ]);
+
+        if (!$input->getOption('force') && $this->ttyDetector->isStdinTty()) {
```
Comment (mitelg) [inline src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: I do not understand fully what the `ttyDetector` is doing, but if we now have a confirmation where was none before, this will break automatic deployment, etc, where this command is used

**Review by mitelg — CHANGES_REQUESTED** [review]

File: src/Core/Framework/Adapter/Command/CacheClearAllCommand.php (hunk 1 above)
Comment (SpiGAndromeda) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: Nothing should change in non-interactive environments. So for CI, cron, etc. everything stays the same.

Comment (mitelg) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: because of this "ttyDetector"?

Comment (mitelg) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: how confident are we about that? never saw a solution like this before. therefore I am very curious :blush: also wondering why there is nothing provided by Symfony for such cases :thinking:

Comment (SpiGAndromeda) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: Symfony has no central class to check TTY. They use the pattern in multiple places: https://github.com/search?q=org%3Asymfony%20stream_isatty&type=code

The tests also document some edge cases documented: https://github.com/shopware/shopware/pull/14544/changes#diff-751187510497df88d641bef1d5ee1ff7522c4dd860aec360f3be1575702f824eR12

"Real" testing of this is diffucult because we dont have simulators or anything similar for different terminals.

Comment (mitelg) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: why don't they came up with a centralized helper for that :angry: 

would you like to make some upstream PR? :grin: :innocent:

Comment (SpiGAndromeda) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: > would you like to make some upstream PR? 😁 😇

Sure, why not. If the PR is merged in symfony, I will add a comment to remember us to replace our implementation.

Comment (shyim) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: why do we need this `--force` flag it makes no sense for me. there is `--no-interaction` by symfony which works automatically for all question confirmation. why do we need to come up with something new.

Even you have your tty is there check, we cannot merge this as it blocks people shells. i am talking about local developer machines with hard-coded scripts/makefiles and so on.

Pov of the community when merged:
image

Comment (SpiGAndromeda) [reply src/Core/Framework/Adapter/Command/CacheClearAllCommand.php]: "local developer machines" ... which would mean they would notice that, read the message once and execute it with `--no-interaction` or just confirm. Its declared a breaking change. Also, we have other commands that require verification like the media cleanup command (sure the impact of this one is bigger, but still).

The `cache:clear:all` command is definitly dangerous and from my experience with agencies and merchants, most of them are unaware what is actually does and that it can cause server errors on user requests (thinking of that, a warning and confirmation in the Admin UI would be good, too).

But restricting the confirmation prompt to production environment would be an alternative. Nuking cache in dev isn't a problem but while logged in via SSH on a production server certainly is.

I looked at the underlying problem and I couldnt think of a solution that doesnt involve incursion into stuff I dont want to touch if its not necessary. My idea was to introduce the plugin hash (not only active ones) as a cache resource to "tag" kernel caches. Running `cache:clear` could then mark sibling kernels as stale and `cache:clear` could be automatically invoked on them with a scheduled task or a simiar deferred mechanics. This would cleanup kernels that should be cleaned up (based on the intention of the `cache:clear` command) while using the neat hot-swap mechanics and prevent downtimes.

You're right on the `--force` flag though.

**Comment by SpiGAndromeda** [comment]
I will close to PR as a documentation update is sufficient for now.
Docs PR: https://github.com/shopware/docs/pull/2088
